src/index.ts

```diff
@@ -1,43 +1,108 @@
 
-interface DateTransformerConfig<T = any> extends CreateAxiosDefaults<T> {
-    // placeholder interface for future configuration
-}
+/**
+ * Configuration options specifically for the Date Transformer.
+ */
+type DateTransformerConfig = {
+    /**
+     * An optional array of strings specifying the field names
+     * that should be converted to Date objects. If provided,
+     * only these fields will be parsed as dates.
+     */
+    allowlist?: string[];
+};
+
+/**
+ * Extended Axios configuration that includes the Date Transformer options.
+ */
+type DateTransformerAxiosConfig<T = any> = CreateAxiosDefaults<T> & DateTransformerConfig;
 
 const dateRegex = /^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(?:\.\d*)?(?:[-+]\d{2}:?\d{2}|Z)?)?$/;
 
-const recursiveDateConversion = (data: any): any => {
+/**
+ * Determines if a given string value matches the ISO date format.
+ * @param value - The value to be checked.
+ * @returns `true` if the value is a valid ISO date string, otherwise `false`.
+ */
+const isDateString = (value: any): boolean => {
+    return dateRegex.test(value);
+};
+
+/**
+ * Checks if a given key is present in the allowlist.
+ * @param key - The key to be checked.
+ * @param allowlist - An optional array of keys to be converted.
+ * @returns `true` if the allowlist is undefined, empty, or if the key is in the allowlist.
+ */
+const isAllowed = (key: string, allowlist?: string[]): boolean => {
+    return !allowlist || allowlist.length === 0 || allowlist.includes(key);
+};
+
+/**
+ * Determines if a given value should be converted to a Date object.
+ * @param key - The key of the value to be converted.
+ * @param value - The value to be checked.
+ * @param allowlist - An optional array of keys that should be converted.
+ * @returns `true` if the value should be converted, otherwise `false`.
+ */
+const shouldConvert = (key: string, value: any, allowlist?: string[]): boolean => {
+    return typeof value === 'string' && isDateString(value) && isAllowed(key, allowlist);
+};
+
+/**
+ * Recursively traverses an object and converts allowed fields to Date objects.
+ * @param data - The data to be traversed and transformed.
+ * @param allowlist - An optional array of keys that should be converted to Date objects.
+ * @returns The transformed data with Date objects where applicable.
+ */
+const recursiveDateConversion = (data: any, allowlist?: string[]): any => {
     if (typeof data === 'object') {
         for (const key in data) {
-            if (typeof data[key] === 'string' && isDateString(data[key])) {
+            if (shouldConvert(key, data[key], allowlist)) {
                 data[key] = new Date(data[key]);
             } else if (typeof data[key] === 'object') {
-                data[key] = recursiveDateConversion(data[key]);
+                data[key] = recursiveDateConversion(data[key], allowlist);
             }
         }
     }
 
     return data;
 };
```

Comment:
the existing recursiveDateConversion logic is functionally correct even if it isn't optimal. Since the tests are passing, it means the logic is robust enough to handle both arrays and objects.